wled00/data/common.js

```diff
@@ -39,16 +37,27 @@ function tooltip(cont=null) {
 			tooltip.style.top = top + "px";
 			tooltip.style.left = left + "px";
 			tooltip.classList.add("visible");
-		});
+}
 
-		element.addEventListener("mouseout", ()=>{
+function hideTooltip(element) {
 			d.querySelectorAll('.tooltip').forEach((tooltip)=>{
 				tooltip.classList.remove("visible");
 				d.body.removeChild(tooltip);
 			});
 			// restore title
 			element.setAttribute("title", element.getAttribute("data-title"));
-		});
+}
+
+function tooltip(cont = null) {
+	const isTouchDevice = 'ontouchstart' in window || navigator.maxTouchPoints > 0;
```

Comment:
This will cause problems on laptops with a touch screen. You won't be able to see the tooltip when you move the mouse over the button.

Maybe do something like this:
```js
var isTouchDevice = false;

window.addEventListener('touchstart', () => {
  isTouchDevice = true;
}, { once: true });

window.addEventListener('mousemove', () => {
  isTouchDevice = false;
}, { once: true });

```

Reply:
Good point, but your solution doesn't work since the event listeners for `touchstart`, `touchend`, `mouseover`, and `mouseout `need to be removed/added. 
I will get back to it when I have a little more time.

Reply:
My code was only intended to replace your `isTouchDevice` calculation.
Your add/remove event listener should still be where you put it.

Reply:
> And you have to consider that on laptops with touchscreens, you can switch between mouse and touch.

I think this would not be a problem since touch input on a laptop with a touchscreen is interpreted as mouse input.

Reply:
> My code snippet was only intended to replace your `isTouchDevice` calculation. Your add/remove event listener should still be where you put it.

Let me clarify what I mean: 
by the time `if (isTouchDevice)` ... is executed, it is very unlikely that the `touchstart` event has already been triggered. 
As a result, only the mouse event listeners will be added, which in practice will lead to the same issue as before.

Here is how I implemented your code snippet:

```
function tooltip(cont = null) {
	var isTouchDevice = false;

	window.addEventListener('touchstart', () => {
	  isTouchDevice = true;
	}, { once: true });
	
	window.addEventListener('mousemove', () => {
	  isTouchDevice = false;
	}, { once: true });

	d.querySelectorAll((cont ? cont + " " : "") + "[title]").forEach((element) => {
		// isTouchDevice is pretty much always false here (except the user somehow manages to trigger the tochstart event in that short time period)
		
		if (isTouchDevice) {
			element.addEventListener("touchstart", () => { showTooltip(element); });
			element.addEventListener("touchend", () => { hideTooltip(element); });
		} else {
			element.addEventListener("mouseover", () => { showTooltip(element); });
			element.addEventListener("mouseout", () => { hideTooltip(element); });
		}
	});
};
```

Reply:
I was thinking of something like this:

```js
var isTouchDevice = false;

window.addEventListener('touchstart', () => {
  isTouchDevice = true;
}, { once: true });

window.addEventListener('mousemove', () => {
  isTouchDevice = false;
}, { once: true });


...


function tooltip(cont = null) {
	d.querySelectorAll((cont ? cont + " " : "") + "[title]").forEach((element) => {
		if (isTouchDevice) {
			element.addEventListener("touchstart", () => { showTooltip(element); });
			element.addEventListener("touchend", () => { hideTooltip(element); });
		} else {
			element.addEventListener("mouseover", () => { showTooltip(element); });
			element.addEventListener("mouseout", () => { hideTooltip(element); });
		}
	});
};
```

Reply:
What do you mean with touch devices? Devices with only a touchscreen or Laptops with touchscreen?

Reply:
devices with only a touch screen